app/views.py

```diff
@@ -0,0 +1,200 @@
+from django.conf import settings
+from django.contrib.auth import authenticate, login, logout
+from django.core.mail import send_mail
+from django.shortcuts import redirect, render
+from django.views.generic import TemplateView, View
+
+from .forms import CommentForm, LoginForms, SignupForms, TaskForm, TaskStatusForm
+from .models import Comment, Task, User
+
+
+# Home page
+class Home(TemplateView):
+    template_name = "index.html"
+
+
+class SignupView(View):
+    template_name = "signup.html"
+
+    def get(self, request):
+        form = SignupForms()
+        return render(request, self.template_name, {"form": form})
+
+    def post(self, request):
+        form = SignupForms(request.POST)
+        if form.is_valid():
+            form.save()
+            return redirect("login")
+        return render(request, self.template_name, {"form": form})
+
+
+class LoginView(TemplateView):
+    template_name = "login.html"
+
+    def get(self, request):
+        if request.user.is_authenticated:
+            return redirect("tasklist")
+        form = LoginForms()
+        return render(request, self.template_name, {"form": form})
+
+    def post(self, request):
+        form = LoginForms(data=request.POST)
+        if form.is_valid():
+            email = form.cleaned_data.get("username")
+            password = form.cleaned_data.get("password")
+            user = authenticate(request, username=email, password=password)
+            if user is not None:
+                login(request, user)
+                return redirect("tasklist")
+        return render(request, self.template_name, {"form": form})
+
+
+class LogoutView(View):
+    def get(self, request):
+        logout(request)
+        return redirect("home_page")
+
+
+class AssignTaskView(View):
+    def get(self, request):
+        users = User.objects.exclude(id=request.user.id)
+        form = TaskForm()
+        return render(
+            request, "assign_task.html", {"form": form, "users": users}
+        )
+
+    def post(self, request):
+        form = TaskForm(request.POST)
+        if form.is_valid():
+            task = form.save(commit=False)
+            task.creator = request.user
+            task.save()
+
+            # Send email
+            assignee = task.assignee
+            subject = f"New Task Assigned: {task.title}"
+            message = f"You have been assigned a new task: {task.title}\nDescription: {task.description}\nPriority: {task.get_priority_display()}\nDue_Date: {task.due_date}"
```

Comment:
create a function in utils.py and use that function here

Reply:
Okay Sir.